### DISCUSSION TITLE
User choice for Authy or Google Authenticator

### DISCUSSION THREAD

**Discussion by ziaulrehman40** [devise-two-factor/devise-two-factor#218] [Q&A]
I am using `devise-two-factor` for 2FA using `google authenticator`, along with backup codes which is working great.

Now I need to *add user choose-able sms based verification(using Authy) alongside*. User should be able to select if user wants to use Google Authenticator or Authy.

I am struggling with basic high level concept/architecture for this scheme. As it seems i will need to monkey patch a lot of stuff if I want to achieve that, even if I try to integrate `devise-authy` alongside. I just want to save some time and document this scheme for anyone who needs this in future.

Any help for this scheme will be great. As i dont seem to find any examples for Authy usage along this gem. I will try to post what worked for me too.
Thanks for the help in advance.

**Comment by john-h-k** [comment]
This is not a thing. You just provide the provisioning URL, both Authy and google auth support it
